ocamltest/main.ml

```diff
@@ -131,8 +137,8 @@ let init_tests_to_skip () =
 let test_file test_filename =
   let start = if Options.show_timings then Unix.gettimeofday () else 0.0 in
   let skip_test = List.mem test_filename !tests_to_skip in
-  let tsl_block = tsl_block_of_file_safe test_filename in
-  let (rootenv_statements, test_trees) = test_trees_of_tsl_block tsl_block in
+  let tsl_ast = tsl_block_of_file_safe test_filename in
+  let (rootenv_statements, test_trees) = test_trees_of_tsl_ast tsl_ast in
   let test_trees = match test_trees with
     | [] ->
```

Comment:
It's in fact worse than that, as there is a difference in behavior between `env` outside the tests and the `env` part of the first node of the test tree, even if there is only one tree, even if we are not running the default tests. I had a few tests failing because of that at some point.

Reply:
This is weird.
1. Can you explain the difference?
2. Do you see it as a part of the specification of ocamltest that should not change, or as an irregularity to be fixed by a future PR?

Reply:
The `Ocaml_actions.find_source_module` function is run on the initial environment before starting to interpret the tree. It is run through the `Environments.intializers` hook system. It seems to be used to compute the `all_modules` variable from a few other variables.

I didn't dig deeply, so I can't say if this would be easy to change.

Reply:
I had a look.

I think that we could get rid of those by avoiding environment initializers: `config_variables` is basically part of the initial empty environment, and `find_source_modules` could be avoided by calling a builtin function to recompute the `all_modules` data whenever it is needed by an action, taking the current environment into account (or using the content of the `all_modules` variable if specified explicitly by the user).

But this is also unimportant and independent of the present change, so I agree with not doing anything about it for now: the initial user-provided environment (the environment-statement prefix before the first test) have a special status, so be it.

Reply:
(This supports the choice of removing the toplevel braces, as the toplevel node has a special status.)